remoteworkitem/trackerquery_repository.go

```diff
@@ -32,208 +30,128 @@ func NewTrackerQueryRepository(db *gorm.DB) *GormTrackerQueryRepository {
-	tq := TrackerQuery{
-		Query:     query,
-		Schedule:  schedule,
-		TrackerID: trackerID,
-		SpaceID:   spaceID,
-	}
-	tx := r.db
-	if err := tx.Create(&tq).Error; err != nil {
-		log.Error(ctx, map[string]interface{}{
-			"err":           err,
-			"tracker_id":    trackerID.String(),
-			"tracker_query": query,
-		}, "unable to create the tracker query")
-		return nil, InternalError{simpleError{err.Error()}}
-	}
-
-	spaceSelfURL := rest.AbsoluteURL(goa.ContextRequest(ctx).Request, app.SpaceHref(spaceID.String()))
-	tq2 := app.TrackerQuery{
-		ID:        strconv.FormatUint(tq.ID, 10),
-		Query:     query,
-		Schedule:  schedule,
-		TrackerID: trackerID,
-		Relationships: &app.TrackerQueryRelationships{
-			Space: app.NewSpaceRelation(spaceID, spaceSelfURL),
-		},
+func (r *GormTrackerQueryRepository) Create(ctx context.Context, tq *TrackerQuery) error {
+	if err := r.db.Create(&tq).Error; err != nil {
+		return errors.NewInternalError(ctx, r.db.Error)
 	}
 
 	log.Info(ctx, map[string]interface{}{
-		"tracker_id":    trackerID.String(),
 		"tracker_query": tq,
 	}, "Created tracker query")
 
-	return &tq2, nil
+	return nil
 }
 
 // Load returns the tracker query for the given id
 // returns NotFoundError, ConversionError or InternalError
-func (r *GormTrackerQueryRepository) Load(ctx context.Context, ID string) (*app.TrackerQuery, error) {
-	id, err := strconv.ParseUint(ID, 10, 64)
-	if err != nil || id == 0 {
-		// treating this as a not found error: the fact that we're using number internal is implementation detail
-		return nil, NotFoundError{"tracker query", ID}
-	}
-
-	log.Info(ctx, map[string]interface{}{
-		"tracker_query_id": id,
-	}, "Loading the tracker query")
-
+func (r *GormTrackerQueryRepository) Load(ctx context.Context, ID uuid.UUID) (*TrackerQuery, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "trackerquery", "load"}, time.Now())
 	res := TrackerQuery{}
-	if r.db.First(&res, id).RecordNotFound() {
+	tx := r.db.Where("id = ?", ID).Find(&res)
+	if tx.RecordNotFound() {
 		log.Error(ctx, map[string]interface{}{
-			"tracker_id": id,
+			"tracker_id": ID.String(),
 		}, "tracker resource not found")
-		return nil, NotFoundError{"tracker query", ID}
+		return nil, errors.NewNotFoundError("tracker query", ID.String())
 	}
-
-	spaceSelfURL := rest.AbsoluteURL(goa.ContextRequest(ctx).Request, app.SpaceHref(res.SpaceID.String()))
-	tq := app.TrackerQuery{
-		ID:        strconv.FormatUint(res.ID, 10),
-		Query:     res.Query,
-		Schedule:  res.Schedule,
-		TrackerID: res.TrackerID,
-		Relationships: &app.TrackerQueryRelationships{
-			Space: app.NewSpaceRelation(res.SpaceID, spaceSelfURL),
-		},
+	if tx.Error != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err":             tx.Error,
+			"trackerquery_id": ID,
+		}, "unable to load the trackerquery by ID")
+		return nil, errors.NewInternalError(ctx, tx.Error)
```

Comment:
You might want to wrap the error. 
```suggestion
		return nil, errors.NewInternalError(ctx, errs.Wrapf(tx.Error, "failed to load the trackerquery by ID: %s", ID))
```

Reply:
done 04eb33c